imsim/telescope_loader.py

```diff
@@ -43,9 +43,111 @@ def parse_xyz(xyz, base):
+    # be supplied.
+    fea:
+      m1m3_gravity:
+        zenith_angle: 30 deg
+
+
+    # Set M1M3 temperature induced figure perturbations.  This requires
+    # the bulk temperature and 4 temperature gradients be supplied.
+    fea:
+      m1m3_temperature:
+        m1m3_TBulk: 0.1  # Kelvin
+        m1m3_TxGrad: 0.01  # Kelvin/meter
+        m1m3_TyGrad: 0.01  # Kelvin/meter
+        m1m3_TzGrad: 0.01  # Kelvin/meter
+        m1m3_TrGrad: 0.01  # Kelvin/meter
+
+    # Engage M1M3 lookup table.  Requires zenith angle and optionally a
+    # fractional random error to apply to each force actuator.
+    fea:
+      m1m3_lut:
+        zenith_angle: 39 deg
+        error: 0.01  # fractional random error to apply to each actuator
+        seed: 1  # random seed for error above
+
+    # Set M2 gravitational perturbations.  Requires zenith angle.
+    fea:
+      m2_gravity:
+        zenith_angle: 30 deg
+
+    # Set M2 temperature gradient induced figure errors.  Requires 2 temperature
+    # gradients (in the z and radial directions).
+    fea:
+      m2_temperature:
+        m2_TzGrad: 0.01  # Kelvin/meter
+        m2_TrGrad: 0.01  # Kelvin/meter
+
+    # Set camera gravitational perturbations.  Requires zenith angle and camera
+    # rotator angle.
+    fea:
+      camera_gravity:
+        zenith_angle: 30 deg
+        rotation_angle: -25 deg
+
+    # Set camera temperature-induced perturbations.  Requires the bulk
+    # temperature of the camera.
+    fea:
+      camera_temperature:
+        camera_TBulk: 0.1
+
+    # Set the Active Optics degrees of freedom.  There are 50 baseline degrees
+    # of freedom, so we won't copy them all here, but you can image a list of
+    # 50 floats as the specifications for each degree of freedom.
+    fea:
+      aos_dof:
+        dof: list-of-50-floats
+
+    Notes
+    -----
+    The implementation of FEA degrees of freedom in ImSim uses the batoid_rubin
+    package.  The available perturbations are algorithmically determined from
+    the `with_*` methods of the LSSTBuilder class there.  The arguments to each
+    `with_*` method are passed in as **kwargs from the config dict.  The only
+    additional processing is that arguments with names ending in `_angle` are
+    parsed using the normal galsim config parsing architecture, which allows one
+    to use unit-ful string "30 deg" or "25.2 arcsec" in addition to a bare
+    float.  Future additions to the `batoid_rubin` package may include new or
+    changed APIs to the examples above.
+    """
+    from batoid_rubin import LSSTBuilder
+    fea_dir = config.pop("fea_dir", "fea_legacy")
+    bend_dir = config.pop("bend_dir", "bend_legacy")
+    builder = LSSTBuilder(telescope, fea_dir=fea_dir, bend_dir=bend_dir)
+    for k, v in config.items():
+        method = getattr(builder, "with_"+k)
+        # parse angles
+        for kk, vv in v.items():
+            if kk.endswith("_angle"):
+                v[kk], safe = ParseValue(v, kk, base, Angle)
```

Comment:
Hmm..  That's not working at the moment b/c I wanted `load_telescope` to also work without a config.  E.g. https://github.com/LSSTDESC/imSim/blob/FEA_perturbations/tests/test_telescope_loader.py#L358.

So base is None and the `ParseValue(v, kk, base, list)` gives 
```
    def GetIndex(config, base, is_sequence=False):
        """Return the index to use for the current object or parameter and the index_key.
    
        First check for an explicit index_key value given by the user.
        Then if base[index_key] is other than obj_num, use that.
        Finally, if this is a sequence, default to 'obj_num_in_file', otherwise 'obj_num'.
    
        Returns:
            index, index_key
        """
        if 'index_key' in config:
            index_key = config['index_key']
            if index_key not in valid_index_keys:
                raise GalSimConfigValueError("Invalid index_key.", index_key, valid_index_keys)
        else:
>           index_key = base.get('index_key','obj_num')
E           AttributeError: 'NoneType' object has no attribute 'get'

```

Reply:
Looks like `ParseValue(v, kk, {}, None)` actually does work, but not sure if there are any other unintended consequences with that hack.

Reply:
Better would be `base = {} is base is None else base` so it would keep the existing base if one is given.